api-spec/api-spec.md

```diff
@@ -50,3 +59,198 @@ The `/stac/search` endpoint is similar to the `items` endpoint in WFS3 in that i
+
+Like Filter Parameters, these attribute names are reserved for extensions, must not implement different semantics.
+
+**TBD** how to allow multiple different extensions format. Partially for allowing multiple to be implemented by the same impl, but more so that you don't accidentally get different semantics because a different format was expected -- need good error handling w/ bad query
+
+| Parameter    | Type             | APIs       | Description        |
+| -----------  | ---------------- | ---------- | ---------------------- |
+| fields | Fields | Placeholder parameter for API Fields Extension. |
+| sort | Sort | Placeholder parameter for API Sort Extensions. |
+| query_type | string | Placeholder parameter for API Query Extension Type (e.g., `cql`). |
+| query | Query | Placeholder parameter for API Query Extension. |
+| operationName | string | Placeholder for GraphQL operationName parameter. |
+| variables | string | Placeholder for GraphQL variables parameter. |
+
+
+
+### Fields fields
+
+This object describes a Fields Extension format.
+
+| Field Name | Type  | Description |
+| ---------- | --------- | ----------- |
+| type       | string    | the Fields Extension name that is used for this, e.g., `include_exclude_csv` |
+| value      | object    | an extension-specific object representing the field filter |
+
+**type** the type as described in the Fields Extension 
+
+**value** the object representing whatever a field definitions means as specified by `type`.  This is completely implementation-specific.
+
+Example:
+
+```
+{ 
+  ...
+  "fields": {
+    "type": "include_exclude_array",
+    "value": {
+      "description": ["id","properties.eo:cloud_cover","properties.proj:geometry","properties.proj:crs"]
+    }
+  }
+  ...
+}
+```
+
+```
+{ 
+  ...
+  "fields": {
+    "type": "include_exclude_csv",
+    "value": {
+      "description": "@summary,-geometry"
+    }
+  }
+  ...
+}
+```
+
+
+### Query fields
+
+This object describes a Query Extension format.
+
+| Field Name | Type  | Description |
+| ---------- | --------- | ----------- |
+| query_type       | string    | the Query Extension name that is used for this, e.g., `cql` |
+| query      | object    | an extension-specific object representing the query filter |
+
+**type** the type as described in the Query Extension 
+
+**value** the object representing whatever a query means in the Query Extension specified by `type`.  This is completely implemenation-specific.
+
+Query Extensions are **recommended** to use attribute names qualified from the root of Item, rather than Item Properties.
+
+Example:
+
+```
+{ 
+  ...
+  "query_type": "cql",
+  "query": {
+    "predicate": "properties.eo:cloud_cover<1 AND properties.landsat:wrs_row=28"
+  }
+  ...
+}
+```
+
+
+```
+{ 
+  ...
+  "type": "existing_query_extension_but_with_arrays",
+  "query": {
+    "predicates": [
+      {
+        "name": "eo:cloud_cover",
+        "gte": 0,
+        "lte": 10
+      }
+    ]
+  }
+  ...
```

Comment:
So this implies that a predicate key could be either "name" or any one of the operators?  It seems to me, if we wanted to keep a syntax such as this, that the predicate should have 3 keys: `field`, `operator` and `value`.  If you wanted items where a field were in some range (cloud cover between 0 and 10) you'd have two predicates, one for less than and one for greater than.  The one thing I still don't like about the syntax is the implicit AN and no support for OR.

Reply:
lol, this is definitely not what i meant to write here.  My fault for doing this late night.  Will update it.

Reply:
I feel strongly that we have a base query extension that is very simple to implement, and I think that means only supporting `AND`.  Lowest barrier to entry possible.